src/data/finset/basic.lean

```diff
@@ -906,6 +906,9 @@ begin
   simp,
 end
 
+lemma erase_inj_on (s : finset α) : set.inj_on s.erase s :=
+λ _ _ _ _, (erase_inj s ‹_›).mp
```

Comment:
Can you swap these proofs around, proving `erase_inj` with `erase_inj_on.eq_iff`? I think it would be slightly shorter.

Reply:
Here's how it would be otherwise:
```
lemma erase_inj_on (s : finset α) : set.inj_on s.erase s :=
λ _ hx _ _ h, eq_of_mem_of_not_mem_erase hx (h ▸ not_mem_erase _ _)

lemma erase_inj {x y : α} (s : finset α) (hx : x ∈ s) :
  s.erase x = s.erase y ↔ x = y :=
begin
  refine ⟨λ h, _, congr_arg _⟩,
  rw [←(erase_inj_on s).eq_iff hx, h],
  contrapose! h,
  rw erase_eq_of_not_mem h,
  exact ne_of_lt (erase_ssubset hx)
end
```